src/apps/bridge/reportBuilder.cpp

```diff
@@ -0,0 +1,490 @@
+        case REPORT_BUILDER_INCREMENT_SAMPLE_COUNT: {
+          _ctx._sample_counter++;
+          printf("Incrementing sample counter to %d\n", _ctx._sample_counter);
+          if (_ctx._sample_counter >= _ctx._samplesPerReport) {
+            if (_ctx._samplesPerReport > 0 && _ctx._transmitAggregations) {
+              // Need to have more than one node to report anything (1 node would be just the bridge)
+              if (_ctx._report_period_num_nodes > 1) {
+                uint32_t num_sensors = _ctx._report_period_num_nodes - 1; // Minus one since topo includes the bridge
+                sensor_report_encoder_context_t context;
+                uint8_t cbor_buffer[315];
+                sensor_report_encoder_open_report(cbor_buffer, sizeof(cbor_buffer), num_sensors, context);
+                // Start at index 1 to skip the bridge
+                for (size_t i = 1; i < _ctx._report_period_num_nodes; i++) {
+                  report_builder_element_t *element = _ctx._reportBuilderLinkedList.findElement(_ctx._report_period_node_list[i]);
+                  if (element == NULL) {
+                    printf("No data for node %" PRIx64 " in report period, adding it to the list\n", _ctx._report_period_node_list[i]);
+                    // we have a sensor that may have been added at the end of the report period
+                    // so lets add it to the list and this wil backfill all of the data so we can still send it
+                    // in the report. We will need to pass (_ctx._sample_counter - 1) to make sure we don't overflow the sensor_data buffer.
+                    // Also we will pass NULL into the sensor_data since we don't have any data for it yet and we will fill the whole thing with NANs
+                    _ctx._reportBuilderLinkedList.findElementAndAddSampleToElement(_ctx._report_period_node_list[i], AANDERAA_SENSOR_TYPE, NULL, sizeof(aanderaa_aggregations_t), _ctx._samplesPerReport, (_ctx._sample_counter - 1));
+                    element = _ctx._reportBuilderLinkedList.findElement(_ctx._report_period_node_list[i]);
+                  } else {
+                    printf("Found data for node %" PRIx64 " adding it the the report\n", _ctx._report_period_node_list[i]);
+                  }
+                  sensor_report_encoder_open_sensor(context, _ctx._samplesPerReport);
+                  for (uint32_t j = 0; j < _ctx._samplesPerReport; j++) {
+                    addSamplesToReport(context, element->sensor_type, element->sensor_data, j);
+                  }
+                  sensor_report_encoder_close_sensor(context);
+                }
+                sensor_report_encoder_close_report(context);
+                size_t cbor_buffer_len = sensor_report_encoder_get_report_size_bytes(context);
+                app_pub_sub_bm_bridge_sensor_report_data_t *message_buff = static_cast<app_pub_sub_bm_bridge_sensor_report_data_t *>(pvPortMalloc(sizeof(uint32_t) + sizeof(size_t) + cbor_buffer_len));
```

Comment:
`sizeof(app_pub_sub_bm_bridge_sensor_report_data_t) + cbor_buffer_len`
instead of summing individual elements.